src/rpc/service_eth.go

```diff
@@ -104,12 +155,52 @@ func (service *EthService) SendRawTransaction(ctx context.Context, s string) (*c
 		return nil, &EncodingError{StatusCode: -32000, Err: errors.New("gas cost is higher")}
 	}
 
-	eon, err := service.processor.KeyperSetManagerContract.GetKeyperSetIndexByBlock(nil, blockNumber+uint64(service.processor.KeyperSetChangeLookAhead))
+	if utils.IsCancellationTransaction(tx, fromAddress) {
+		utils.Logger.Info().Msg("Detected cancellation transaction, forwarding to backend")
+
+		backendClient, err := rpc.Dial(service.Config.BackendURL.String())
+		if err != nil {
+			utils.Logger.Err(err).Msg("Failed to connect to backend")
+			return nil, &EncodingError{StatusCode: -32603, Err: err}
+		}
+
+		err = backendClient.CallContext(ctx, &txHash, "eth_sendRawTransaction", s)
+		if err != nil {
+			utils.Logger.Err(err).Msg("Failed to send cancel transaction to backend")
+			return nil, &EncodingError{StatusCode: -32602, Err: err}
+		}
+
+		utils.Logger.Info().Msg("Transaction forwarded with hash: " + txHash.Hex())
+		return &txHash, nil
+	}
+
+	updated, err := service.Cache.UpdateEntry(tx, uint64(time.Now().Unix()))
```

Comment:
Is there ever gonna be the intention to pass other values than `uint64(time.Now().Unix())`? If not then it should be  moved into the method saving us the additional argument

Reply:
updated to int64